src/datachain/query/dataset.py

```diff
@@ -1021,23 +1018,46 @@ def q(*columns):
         return step_result(q, grouped_query.selected_columns)
 
 
-def fill_columns(
-    *column_iterables: Iterable[ColumnElement],
+def _validate_columns(
+    left_columns: Iterable[ColumnElement], right_columns: Iterable[ColumnElement]
+) -> set[str]:
+    left_names = {c.name for c in left_columns}
+    right_names = {c.name for c in right_columns}
+
+    if left_names == right_names:
+        return left_names
+
+    missing_right = left_names - right_names
+    missing_left = right_names - left_names
+
+    def _prepare_msg_part(missing_columns: set[str], side: str) -> str:
+        return f"{', '.join(sorted(missing_columns))} only present in {side}"
+
+    msg_parts = [
+        _prepare_msg_part(missing_columns, found_side)
+        for missing_columns, found_side in zip(
+            [
+                missing_right,
+                missing_left,
+            ],
+            ["left", "right"],
+        )
+        if missing_columns
+    ]
+    msg = f"Cannot perform union. {'. '.join(msg_parts)}"
```

Comment:
Would it make sense to have a period (full stop) at the end of this message always? It looks a bit odd to have a period at the end of "Cannot perform union." but not at the end of the message, unless there are missing columns on both sides, where there is a period in between these two statements as well, but not at the end.

Reply:
I can update but I was trying to be consistent with the bulk of the other error messages in the repo. Generally the "standard" seems to be a capital first letter and no period at the end of the error message. WDYT?

Reply:
I'm not really sure if it's an official standard, or just something we've been doing. It's totally fine keeping the current way, it just looked a bit odd with the other periods in the message, but not at the end, although it doesn't really matter much to me. (Possibly I only noticed this due to the other periods, which we usually don't have.)